### DISCUSSION TITLE
Can you make the scripts run faster and earlier?

### DISCUSSION THREAD

**Discussion by NoobTracker** [ScratchAddons/ScratchAddons#159] [Help]
The addon is completely awesome, no question. But a little thing that annoys me is that the discuss button only loads after a second. Same thing goes for my forkphorus button. Is it possible to make that faster?

**Comment by Explosion-Scratch** [comment]
Sorry but probably not. =(

**Comment by NoobTracker** [comment]
Why exactly does it run so lately? I messed a bit with a forum to add some stuff and that's instantly there.

**Comment by apple502j** [comment]
@NoobTracker Userscripts run at `onload` instead of `DOMContentLoaded`. We may change this in the future.

**Comment by NoobTracker** [comment]
Okay, thanks for the information, is there a reason why it isn't already changed?

**Comment by apple502j** [comment]
Not sure but I think some extensions rely on current behaviors.

**Comment by NoobTracker** [comment]
Where is that line of code?

Maybe we could solve the problem by adding 
`"DOMContentLoaded": true`
to the JSONs where it makes sense and nothing is treated as false.

**Comment by apple502j** [comment]
We need to modify cs.js and stuff. It's not just a line of code.

**Comment by NoobTracker** [comment]
What exactly causes that to be executed `onload`?

**Comment by apple502j** [comment]
`manifest.json` specifies `cs.js` to be loaded at `document_idle`, which means `onload`.
NOTE: We want some existing extensions to load at `document_idle`, so please do not make PR.

**Comment by NoobTracker** [comment]
Ah okay.
No, didn't want to do that PR. Maybe with the `DOMContentLoaded` JSON thing but not just by changing that thing. 
How can you change that? What do I have to insert instead of `document_idle`?

**Comment by WorldLanguages** [comment]
> Userscripts run at `onload` instead of `DOMContentLoaded`. We may change this in the future.

Actually, remember addon scripts are JS modules, which are said to always run in `defer` mode, AKA when the page has loaded. However, I'm not sure if it's possible to use dynamic `import()` to bypass that restriction.
